### PR TITLE
refactor synapse dedicated sqlpool  [sc-14405]

### PR DESCRIPTION
**Pull request by pk1230h**
### 🤔 Why?
refactor synapse dedicate sqlpool from API call to use system catalog views

### 🤓 What?
1. remove dedicated SQL API and use system catalog views
2. remove unused  API config
3. adjust readme

### 🧪 Tested?
unit test and manual test

### CODE REVIEW

File: metaphor/synapse/extractor.py [hunk 1]
```diff
@@ -85,15 +77,15 @@ async def extract(self) -> Collection[ENTITY_TYPES]:
 
         # Dedicated sqlpool
         try:
-            for database in workspaceClient.get_dedicated_sql_pool_databases():
-                tables = workspaceClient.get_dedicated_sql_pool_tables(database.name)
-                datasets = self._map_dedicated_sql_pool_tables_to_dataset(
-                    workspaceClient._workspace, database, tables
+            for database in self._client.get_databases(True):
```
Comment (elic-eon) [inline metaphor/synapse/extractor.py:80]: Specify `is_dedicated` explicitly?

File: metaphor/synapse/extractor.py [hunk 2]
```diff
@@ -85,15 +77,15 @@ async def extract(self) -> Collection[ENTITY_TYPES]:
...
+                tables = self._client.get_tables(database.name, True)
```
Comment (elic-eon) [inline metaphor/synapse/extractor.py:81]: Also here

**Review by elic-eon — COMMENTED** [review]
Otherwise, LGTM

File: metaphor/synapse/extractor.py (hunk 1 above)
Comment (pk1230h) [reply metaphor/synapse/extractor.py]: fixed

File: metaphor/synapse/extractor.py (hunk 2 above)
Comment (pk1230h) [reply metaphor/synapse/extractor.py]: fixed

**Review by elic-eon — APPROVED** [review]

**Review by elic-eon — CHANGES_REQUESTED** [review]
Bump version

**Review by elic-eon — APPROVED** [review]

**Pull request merged**